src/index.ts

```diff
@@ -1,33 +1,31 @@
-/**
- * Welcome to Cloudflare Workers! This is your first worker.
- *
- * - Run `npm run dev` in your terminal to start a development server
- * - Open a browser tab at http://localhost:8787/ to see your worker in action
- * - Run `npm run deploy` to publish your worker
- *
- * Learn more at https://developers.cloudflare.com/workers/
- */
-
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface Env {
-  // Example binding to KV. Learn more at https://developers.cloudflare.com/workers/runtime-apis/kv/
-  // MY_KV_NAMESPACE: KVNamespace;
-  //
-  // Example binding to Durable Object. Learn more at https://developers.cloudflare.com/workers/runtime-apis/durable-objects/
-  // MY_DURABLE_OBJECT: DurableObjectNamespace;
-  //
-  // Example binding to R2. Learn more at https://developers.cloudflare.com/workers/runtime-apis/r2/
-  // MY_BUCKET: R2Bucket;
-  //
-  // Example binding to a Service. Learn more at https://developers.cloudflare.com/workers/runtime-apis/service-bindings/
-  // MY_SERVICE: Fetcher;
-  //
-  // Example binding to a Queue. Learn more at https://developers.cloudflare.com/queues/javascript-apis/
-  // MY_QUEUE: Queue;
-}
-
-export default {
-  async fetch (request: Request, env: Env, ctx: ExecutionContext): Promise<Response> {
-    return new Response('Hello World!')
-  }
-}
+import { Env } from "./env";
+import { Hono } from "hono";
+import { Service } from "./types";
+
+const app = new Hono<{ Bindings: Env }>();
+app
+  .all("*", async (c) => {
+    const xService = c.req.raw.headers.get("x-service");
+
+    if (!xService) {
+      return c.newResponse("x-service header is required", { status: 400 });
+    }
+
+    const service = await c.env.SERVICES.get<Service>(xService, { type: "json" });
+
+    if (!service) {
+      return c.newResponse("Service not found", { status: 500 });
+    }
+
+    console.log(service.host, service.token);
+
+    const url = new URL(service.host);
+    url.pathname = new URL(c.req.raw.url).pathname;
+    url.search = new URL(c.req.raw.url).searchParams.toString();
+    const headers = new Headers(c.req.raw.headers);
+    headers.append("Authorization", `Bearer ${service.token}`);
```

Comment:
Some of the API's don't use auth header. Will most likely need another header to check what to use.

Reply:
I think all of them use request parameters, so I'll just list them here.  
I don't find any service that uses `Authorization` header, so I'll just ignore it.

1. **Google Safe Browsing API (webrisk.googleapis.com):**
    - **Query Parameter:**
      ```
      key=your_api_key_here
      ```

2. **WhoisXMLAPI (www.whoisxmlapi.com, user.whoisxmlapi.com, domain-reputation.whoisxmlapi.com, domain-availability.whoisxmlapi.com, ip-geolocation.whoisxmlapi.com, website-categorization.whoisxmlapi.com, website-contacts.whoisxmlapi.com, emailverification.whoisxmlapi.com):**
    - **Query Parameter:**
      ```
      apikey=your_api_key_here
      ```

3. **Monapi (api.monapi.io):**
    - **Query Parameter:**
      ```
      userKey=your_api_key_here
      ```

4. **Ipify (api64.ipify.org, geo.ipify.org, vpn-proxy-detection.ipify.org):**
    - **Query Parameter:**
      ```
      apiKey=your_api_key_here
      ```

5. **Shodan (api.shodan.io):**
    - **Query Parameter:**
      ```
      key=your_api_key_here
      ```

I'll just append the respective query parameter to the end of the URL, and then forward the request to the original server.

Reply:
>I don't find any service that uses Authorization header, so I'll just ignore it.

OpenAI does

Reply:
yeah, missed that one 😅.